curvefs/test/mds/mds_service_test.cpp

```diff
@@ -128,6 +143,154 @@ class MdsServiceTest : public ::testing::Test {
             fsStorage_, spaceManager_, metaserverClient_, topoManager_,
             s3Adapter_, nullptr, fsManagerOption);
         ASSERT_TRUE(fsManager_->Init());
+
+
+        mdsService_ = std::make_shared<MdsServiceImpl>(fsManager_, nullptr);
+        // add metaserver service
+        ASSERT_EQ(server.AddService(mdsService_.get(),
+                  brpc::SERVER_DOESNT_OWN_SERVICE), 0);
+
+        FakeMetaserverImpl metaserverService;
+        ASSERT_EQ(server.AddService(&metaserverService,
+                                    brpc::SERVER_DOESNT_OWN_SERVICE), 0);
```

Comment:
In theory, it will result in invalid memory access, because `metaserverService` is located on stack, and space for it will be recycled after function returned, you can refer to `mockCliService2` for modifications.

Reply:
relocated